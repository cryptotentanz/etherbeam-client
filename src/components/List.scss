@use '../styles/resources/sizes';

.my-list-h,
.my-list-v {
  display: flex;
  flex-wrap: wrap;
  margin: -#{sizes.$spacer-xs};
  padding: 0px;

  .my-listitem {
    margin: sizes.$spacer-xs;
  }
}

.my-list {
  &-h {
    flex-direction: row;
    align-items: center;
  }

  &-v {
    flex-direction: column;
  }

  &-center {
    justify-content: center;
  }
}
