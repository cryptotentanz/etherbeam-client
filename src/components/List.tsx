import React, { FC, ReactNode } from 'react'
import './List.scss'

interface ListProps {
  horizontal?: boolean
  center?: boolean
  className?: string
  children?: ReactNode
}

const List: FC<ListProps> = ({ horizontal, center, className, children }: ListProps) => {
  const ulClassName = [horizontal ? 'my-list-h' : 'my-list-v', center ? 'my-list-center' : '', className].join(' ')

  return <ul className={ulClassName}>{children}</ul>
}

export default List
