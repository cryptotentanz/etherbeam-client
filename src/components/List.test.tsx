import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import List from './List'

describe('<List />', () => {
  const content = 'Content'
  const className = 'my-class'

  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<List>{content}</List>)
    })

    it('should render', () => {
      const ul = component.find('ul')
      expect(ul.prop('children')).toBe(content)
      expect(ul.hasClass('my-list-v')).toBeTruthy()
      expect(ul.hasClass('my-list-h')).toBeFalsy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Horizontal', () => {
    beforeAll(() => {
      component = shallow(<List horizontal>{content}</List>)
    })

    it('should render', () => {
      const ul = component.find('ul')
      expect(ul.prop('children')).toBe(content)
      expect(ul.hasClass('my-list-v')).toBeFalsy()
      expect(ul.hasClass('my-list-h')).toBeTruthy()
    })
  })

  describe('Center', () => {
    beforeAll(() => {
      component = shallow(<List center>{content}</List>)
    })

    it('should render', () => expect(component.find('ul').hasClass('my-list-center')).toBeTruthy())
  })

  describe('Class', () => {
    beforeAll(() => {
      component = shallow(<List className={className}>{content}</List>)
    })

    it('should render', () => expect(component.find('ul').hasClass(className)).toBeTruthy())
  })
})
