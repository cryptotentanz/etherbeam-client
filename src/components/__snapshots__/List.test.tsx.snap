// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Default should match snapshot 1`] = `
<ul
  class="my-list-v  "
>
  Content
</ul>
`;
